app/src/main/java/au/com/shiftyjelly/pocketcasts/ui/MainActivity.kt

```diff
@@ -752,22 +753,23 @@ class MainActivity :
             }
         }
 
-        val upNextQueueObservable =
-            playbackManager.upNextQueue.getChangesObservableWithLiveCurrentEpisode(
-                episodeManager,
-                podcastManager,
-            )
-                .subscribeOn(Schedulers.io())
-                .observeOn(AndroidSchedulers.mainThread())
-                .toFlowable(BackpressureStrategy.LATEST)
-        observeUpNext = upNextQueueObservable.toLiveData()
-        observeUpNext.observe(this) { upNext ->
-            binding.playerBottomSheet.setUpNext(
-                upNext = upNext,
-                theme = theme,
-                shouldAnimateOnAttach = !showMiniPlayerImmediately,
-            )
+        val upNextQueueChanges = playbackManager.upNextQueue.getChangesFlowWithLiveCurrentEpisode(episodeManager, podcastManager)
+        val useRssArtworkChanges = settings.useRssArtwork.flow
+
+        combine(upNextQueueChanges, useRssArtworkChanges) { upNextQueue, useRssArtwork ->
+            Timber.i("useRssArtwork combined, $useRssArtwork")
+            upNextQueue to useRssArtwork
         }
+            .onEach { (upNextQueue, useRssArtwork) ->
+                binding.playerBottomSheet.setUpNext(
+                    upNext = upNextQueue,
+                    theme = theme,
+                    shouldAnimateOnAttach = !showMiniPlayerImmediately,
+                    useRssArtwork = useRssArtwork
+                )
+            }
+            .catch { Timber.e(it) }
+            .launchIn(this)
```

Comment:
Oh, I completely missed that part in the review. We shouldn't be using implicit `CoroutineScope` in the activities. In this particular instance it would cause memory leaks.

Instead please use `repeatOnLifecycle` API. It would look something like this. 

```kotlin
lifecycleScope.launch {
  repeatOnLifecycle(Lifecycle.State.STARTED) {
    <your_flow>.collect()
  }
}
```